src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_northrend_beasts.cpp

```diff
@@ -1151,6 +1171,141 @@ class boss_icehowl : public CreatureScript
+        }
+
+        void Register() override
+        {
+            AfterEffectRemove += AuraEffectRemoveFn(spell_gormok_jump_to_hand_AuraScript::OnRemove, EFFECT_0, SPELL_AURA_CONTROL_VEHICLE, AURA_EFFECT_HANDLE_REAL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_gormok_jump_to_hand_AuraScript();
+    }
+};
+
+class spell_gormok_ride_player : public SpellScriptLoader
+{
+public:
+    spell_gormok_ride_player() : SpellScriptLoader("spell_gormok_ride_player") { }
+
+    class spell_gormok_ride_player_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_gormok_ride_player_AuraScript);
+
+        bool Load() override
+        {
+            if (GetCaster() && GetCaster()->GetEntry() == NPC_SNOBOLD_VASSAL)
+                return true;
+            return false;
+        }
+
+        void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            Unit* target = GetTarget();
+
+            if (target->GetTypeId() != TYPEID_PLAYER || !target->IsInWorld())
+                return;
+
+            if (!target->CreateVehicleKit(PLAYER_VEHICLE_ID, 0))
+                return;
+
+            if (Unit *caster = GetCaster())
+                caster->GetAI()->SetGUID(target->GetGUID(), DATA_NEW_TARGET);
+        }
+
+        void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            Unit* target = GetTarget();
+            target->RemoveAurasDueToSpell(SPELL_SNOBOLLED);
+            target->RemoveVehicleKit();
+        }
+
+        void Register() override
+        {
+            OnEffectApply += AuraEffectApplyFn(spell_gormok_ride_player_AuraScript::OnApply, EFFECT_0, SPELL_AURA_CONTROL_VEHICLE, AURA_EFFECT_HANDLE_REAL);
+            AfterEffectRemove += AuraEffectRemoveFn(spell_gormok_ride_player_AuraScript::AfterRemove, EFFECT_0, SPELL_AURA_CONTROL_VEHICLE, AURA_EFFECT_HANDLE_REAL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_gormok_ride_player_AuraScript();
+    }
+};
+
+class spell_gormok_snobolled : public SpellScriptLoader
+{
+public:
+    spell_gormok_snobolled() : SpellScriptLoader("spell_gormok_snobolled") { }
+
+    class spell_gormok_snobolled_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_gormok_snobolled_AuraScript);
+
+        bool Validate(SpellInfo const* /*spell*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_RIDE_PLAYER))
+                return false;
+            return true;
+        }
+
+        void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (!GetTarget()->HasAura(SPELL_RIDE_PLAYER))
+                Remove();
+        }
+
+        void Register() override
+        {
+            OnEffectApply += AuraEffectApplyFn(spell_gormok_snobolled_AuraScript::OnApply, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
```

Comment:
It's `PERIODIC_DUMMY`, so maybe it does the removal check in its periodic handler (instead of getting removed on afterremove of snowbolled)?

Not sure it makes any difference, but there's usually a reason why retail uses `_PERIODIC_`.